sklearn/base.py

```diff
@@ -562,3 +561,37 @@ def is_regressor(estimator):
         True if estimator is a regressor and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "regressor"
+
+
+def is_pairwise(estimator):
+    """Returns True if estimator has a _pairwise attribute set to True.
+
+    Parameters
+    ----------
+    estimator : object
+        Estimator object to test.
+
+    Returns
+    -------
+    out : bool
+        True if _pairwise is set to True and False otherwise.
+    """
+    return bool(getattr(estimator, "_pairwise", False))
+
+
+def is_pairwise_metric(estimator):
+    """Returns True if estimator accepts pairwise metric.
```

Comment:
That doesn't seem right. If the metric is set to ``precomputed``, right? I'm not sure I understand the context.

Reply:
Ah, I think I got it. This tests whether we accept pairwise distances, as opposed to pairwise similarities? That's not really clear to me from the description.